systemFlake.nix

```diff
@@ -48,34 +53,42 @@ let
     "checksBuilder"
   ];
 
-  nixosConfigurationBuilder = hostname: profile: (
-    # It would be nice to get `nixosSystem` reference from `selectedNixpkgs` but it is not possible at this moment
-    inputs."${channelNameFromProfile profile}".lib.nixosSystem (genericConfigurationBuilder hostname profile)
-  );
-
   getNixpkgs = profile: self.pkgs."${systemFromProfile profile}"."${channelNameFromProfile profile}";
 
-  genericConfigurationBuilder = hostname: profile: (
-    let selectedNixpkgs = getNixpkgs profile; in
+  configurationBuilder = hostname: profile: (
+    let
+      selectedNixpkgs = getNixpkgs profile;
+      defaultBuilder =
+        if defaultConfigBuilder == null
+        then inputs."${channelNameFromProfile profile}".lib.nixosSystem
+        else defaultConfigBuilder;
```

Comment:
keep the nice nixos support, while still not sacrificing convenience for non-nixos.